example/oven_fsm.dart

```diff
@@ -0,0 +1,126 @@
+import 'package:rohd/rohd.dart';
+
+import './counter.dart';
+
+enum OvenStates { standby, cooking, paused, completed }
+
+class Button extends Const {
+  Button._(int super.value) : super(width: 2);
+  Button.start() : this._(bin('00'));
+  Button.pause() : this._(bin('01'));
+  Button.resume() : this._(bin('10'));
+}
+
+class LEDLight extends Const {
+  LEDLight._(int super.value) : super(width: 2);
+  LEDLight.yellow() : this._(bin('00'));
+  LEDLight.blue() : this._(bin('01'));
+  LEDLight.red() : this._(bin('10'));
+  LEDLight.green() : this._(bin('11'));
+}
+
+class OvenModule extends Module {
+  late Logic clk;
+  late Logic reset;
+  late List<State<OvenStates>> states;
+
+  OvenModule(Logic button, Logic reset) : super(name: 'OvenModule') {
+    // input to FSM
+    button = addInput('button', button, width: button.width);
+
+    // output to FSM
+    final led = addOutput('led', width: button.width);
+
+    // add clock & reset
+    clk = SimpleClockGenerator(10).clk;
+    this.reset = addInput('reset', reset);
+
+    // add time elapsed Counter
+    final counterReset = Logic(name: 'counter_reset');
+    final en = Logic(name: 'counter_en');
+    final counter = Counter(en, counterReset, clk, name: 'counter_module');
+
+    states = [
+      // Standby State
+      State<OvenStates>(OvenStates.standby, events: {
+        button.eq(Button.start()): OvenStates.cooking,
```

Comment:
Let's try naming the `Logic`s that we pass into there so that the FSM diagram has some more interesting stuff?
```dart
Logic(name: 'button_start')..gets(button.eq(Button.start()))
```

Reply:
Yup, sure!